cpp/tests/types/dispatcher_test.cu

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2018, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cstdint>
+#include "gtest/gtest.h"
+#include <types/type_dispatcher.hpp>
+#include <thrust/device_vector.h>
+#include <cudf.h>
+#include "tests/utilities/cudf_test_fixtures.h"
+
+/**
+ * @file dispatcher_test.cu
+ * @brief Tests the type_dispatcher
+*/
+
+namespace{ 
+
+struct test_functor {
+  template <typename T> 
+  __host__ __device__
+  bool operator()(gdf_dtype type_id) {
+    switch (type_id) {
+    case GDF_INT8:
+      return (std::is_same<T, int8_t>::value);
+    case GDF_INT16:
+      return (std::is_same<T, int16_t>::value);
+    case GDF_INT32:
+      return (std::is_same<T, int32_t>::value);
+    case GDF_INT64:
+      return (std::is_same<T, int64_t>::value);
+    case GDF_FLOAT32:
+      return (std::is_same<T, float>::value);
+    case GDF_FLOAT64:
+      return (std::is_same<T, double>::value);
+    case GDF_CATEGORY:
+      return (std::is_same<T, cudf::category>::value);
+    case GDF_TIMESTAMP:
+      return (std::is_same<T, cudf::timestamp>::value);
+    case GDF_DATE32:
+      return (std::is_same<T, cudf::date32>::value);
+    case GDF_DATE64:
+      return (std::is_same<T, cudf::date64>::value);
+    default:
+      return (false);
+    }
+  }
+};
+
+__global__ 
+void dispatch_test_kernel(gdf_dtype type, bool * d_result)
+{
+  if(0 == threadIdx.x + blockIdx.x * blockDim.x)
+    *d_result = cudf::type_dispatcher(type, test_functor{}, type);
+}
+
+} // anonymous namespace
+
+struct DispatcherTest : public GdfTest 
+{
+  std::vector<gdf_dtype> dtype_enums{GDF_INT8, GDF_INT16, GDF_INT32, GDF_INT64, GDF_FLOAT32, GDF_FLOAT64, GDF_DATE32, GDF_DATE64, GDF_TIMESTAMP, GDF_CATEGORY};
```

Comment:
Please wrap at 80 columns (not readable without horizontal scrolling on GitHub, among all other historical reasons to do so).